include/boost/geometry/algorithms/detail/is_valid/has_duplicates.hpp

```diff
@@ -48,7 +48,8 @@ struct has_duplicates
         geometry::equal_to<typename boost::range_value<Range>::type> equal;
 
         iterator it = boost::begin(view);
-        iterator next = ++boost::begin(view);
+        iterator next = it;
+        ++next;
```

Comment:
And then the line 52 would fail from the reason I described before. For this to work also `iterator next` type should be changed to `const_iterator`.